lms/static/js/learner_dashboard/models/course_enroll_model.js

```diff
@@ -0,0 +1,97 @@
+/**
+ *  Track optIn status and submit form
+ */
+;(function (define) {
+    'use strict';
+
+    define([
+            'backbone',
+            'jquery',
+            'underscore'
+        ],
+        function( Backbone, $, _ ) {
+            return Backbone.Model.extend({
+                defaults: {
+                    course_id: '',
+                    optIn: false,
+                    enrolled: false,
+                },
+
+                initialize: function( attributes, options ) {
+                    var api = '/api/commerce/v0/baskets/';
+
+                    this.set({
+                        course_id: options.courseId,
+                    });
+
+                    this.url = api;
+                    this.trackSelectionUrl = '/course_modes/choose/';
+                    this.dashboardUrl = '/dashboard';
+                    this.verificationUrl = '/verify_student/start-flow/';
+                    this.csrfCookieName = 'csrftoken';
+                },
+
+
+                /**
+                 *  Enroll in a course
+                 */
+                enroll: function( track ) {
+                    var data = {
+                            course_id: this.get('course_id'),
+                            email_opt_in: this.get('optIn')
+                        };
+
+                    $.ajax({
```

Comment:
It would be nice if we could use the inbuilt ```save``` function (http://backbonejs.org/#Model-save).